### ISSUE TITLE
Network error with HTTPS endpoints

### ISSUE THREAD

**Issue by znmeb** [hackoregon/2019-transportation-systems-backend#63]
How to reproduce:

1. Browse to `https://service.civicpdx.org/transportation2019/v1/schema/`
2. Open the `toad` list.
3. Open `/transportation2019/v1/toad/busSystemWideSummary/`
4. Press "Try it out" and "Execute".

![Screenshot_2019-08-27 Swagger UI](https://user-images.githubusercontent.com/4938/63802129-c8d51b80-c8c6-11e9-8457-9d5a1df65db0.png)

This works if I change `https` to `http` in step 1.

**Comment by nam20485** [comment]
https://github.com/tomakehurst/wiremock/issues/965

https://github.com/swagger-api/swagger-ui/issues/4710

**Comment by karenng-civicsoftware** [comment]
Potential fix: we should edit 
https://github.com/hackoregon/2019-backend-cookiecutter-django/blob/master/%7B%7Bcookiecutter.github_repo%7D%7D/local_settings/settings.py 
and the likes according to 
https://github.com/marcgibbons/django-rest-swagger/issues/363#issuecomment-146753452

Just add the line `SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTOCOL', 'https')`

**Comment by bhgrant8** [comment]
> Potential fix: we should edit
> https://github.com/hackoregon/2019-backend-cookiecutter-django/blob/master/%7B%7Bcookiecutter.github_repo%7D%7D/local_settings/settings.py
> and the likes according to
> [marcgibbons/django-rest-swagger#363 (comment)](https://github.com/marcgibbons/django-rest-swagger/issues/363#issuecomment-146753452)
> 
> Just add the line `SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTOCOL', 'https')`

We are not running nginx though, so not sure if we are running a proxy? there are some definite caveats on this in the django docs:
https://docs.djangoproject.com/en/2.2/ref/settings/#secure-proxy-ssl-header

https://github.com/hackoregon/civic-devops/issues/210

I might think first starting with the ssl redirect, might be fruitful test?

https://docs.djangoproject.com/en/2.2/ref/settings/#secure-ssl-redirect

**Comment by nam20485** [comment]
>Just add the line SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTOCOL', 'https')

Brian's conclusions are correct. There is definitely more involved than just adding the settings parameter outlined in the proposed solution.

For one thing, we do not have an SSL certificate.

**Comment by znmeb** [comment]
What do we need to do to get a certificate? Is "Let's encrypt" applicable? https://letsencrypt.org/

**Comment by nam20485** [comment]
I think that we need to research how to properly enable https/ssl and and then create an issue that outlines all the necessary changes so that we can understand the scope of this request.

The last two times we looked at this it looked like a non-trivial endeavor.

I don't know anything about enabling https in django, but I did read that not enabling it fully correctly can introduce more security vulnerabilities than simply not doing it all, which is why I think we need to understand the scope of the changes required.

Do we have anybody in-house that has experience enabling https in django previously?
